cmd/shim/main.go

```diff
@@ -41,6 +41,11 @@ import (
 	"github.com/dagger/dagger/telemetry"
 )
 
+func init() {
+	// unset umask so permissions are created as specified
+	unix.Umask(0)
```

Comment:
This is needed because previously we had a umask of `0o22`, which resulted in any files created here never giving `w` perms to group+other, even if created as `0o666`. This became a problem now because if we need to run ca certs update commands as root initially but the final container used a non-root user, then the shim failed at trying to open the `stdout/stderr` files for write.

There may be other ways of addressing this but afaict there shoudn't be any harm in unsetting the umask here. The only thing to double check still is whether the umask here actually propagates to the final container init process, in which case this would be a breaking change (*technically*) and is probably better avoid.

Reply:
Double-checked this, `umask` is actually in oci runtime spec so it's important we honor it. So will need a different approach; I think we could just set the owner of those stdout/stderr files to be that of the final container uid instead, which should be relatively simple still.

Reply:
Fixed here https://github.com/dagger/dagger/pull/7067/commits/0591a1ff832fd8dd385ff6197c0379363284fe4d